eclass/kernel-build.eclass

```diff
@@ -114,6 +114,15 @@ kernel-build_pkg_setup() {
 	python-any-r1_pkg_setup
 	if [[ ${KERNEL_IUSE_MODULES_SIGN} ]]; then
 		secureboot_pkg_setup
+		if [[ -e ${MODULES_SIGN_KEY} && -e ${MODULES_SIGN_CERT} &&
+			${MODULES_SIGN_KEY} != ${MODULES_SIGN_CERT} &&
+			${MODULES_SIGN_KEY} != pkcs11:* ]]
+		then
+			cat "${MODULES_SIGN_CERT}" "${MODULES_SIGN_KEY}" > "${T}/kernel_key.pem" || die
+			chmod 0600 "${T}/kernel_key.pem" || die
```

Comment:
I wonder if it makes sense to do `0400` instead, we only need to read after all.

Reply:
> I wonder if it makes sense to do `0400` instead, we only need to read after all.

I tried doing that, but for whatever reason the perms ended up being 0600 regardless. I need to look into it further.

Reply:
I suspect this happens because this does chmod before chown. If the file is still owned by root then 0400 permissions don't make sense because root can always read and write. Probably it will work if you switch the order of chmod/chown.